testplan/testing/multitest/base.py

```diff
@@ -802,9 +806,25 @@ def _execute_step(self, step, *args, **kwargs):
         with self.report.logged_exceptions():
             try:
                 res = step(*args, **kwargs)
+                if (
+                    self.resources.start_exceptions
+                    or self.resources.stop_exceptions
+                ):
+                    self.logger.critical(
+                        'Found exceptions during "%s" step', step.__name__
+                    )
+                    if self.cfg.error_handler:
+                        self.logger.critical("Running error handler!")
+                        self._run_error_handler(self.cfg.error_handler, step)
```

Comment:
_run_error_handler is always running the self.cfg.error_handler, no need to get it as a parameter. You might be move the test of a valid error handfler and the logging into the function.